### PR TITLE
Hype Detector API

### PR DESCRIPTION
**Pull request by pdamodaran**
This PR includes the following:
- A scoring API to detect hype
- A demo streamlit app 
- Updates to the README to include instructions for installing black
- Updates to requirements.txt for additional packages required by the hype detector demo (we may want to split out the requirements file in the future)

The demo app currently loads the model from disk but it will eventually load the model from Ensign.  I will include that functionality in a future PR.

### CODE REVIEW

File: halcyon/hype_detector.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        Returns a score in the range [0, 1] that is a representation of
+        the number of hashtags, exclamation points, and emojis used in
+        a post.  This scores gives further insight into how much a post
+        is hyped up by the use of hashtags, emojis, and exclamation points
+        to capture attention.
+        """
+        ht_count = text.count("#")
+        exclamation_count = text.count("!")
+        emoji_count = emoji.emoji_count(text)
```
Comment (pdeziel) [inline halcyon/hype_detector.py:57]: Nice, the emoji count is a good idea and it's great that there's a package for it.

File: halcyon/hype_detector.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    def clean_text(self, text: str) -> str:
+        """
+        Strip out common text
+        """
+        all_patterns = [
+            "…see more",
+            "https://lnkd.in/",
+            "www.bit.ly/",
+            "http://bit.ly",
+            "#",
```
Comment (pdeziel) [inline halcyon/hype_detector.py:41]: Are the hashtags getting filtered out before the hype score?

**Review by pdeziel — APPROVED** [review]
Looks great, I like that we're keeping the API interfaces pretty consistent with each other.

File: halcyon/hype_detector.py (hunk 2 above)
Comment (pdamodaran) [reply halcyon/hype_detector.py:41]: Good question!  I am using the original content field to compute the hype score.

**Pull request merged**